test/dbservice_web/controllers/batch_controller_test.exs

```diff
@@ -0,0 +1,101 @@
+defmodule DbserviceWeb.BatchControllerTest do
+  use DbserviceWeb.ConnCase
+
+  import Dbservice.BatchesFixtures
+
+  alias Dbservice.Batches.Batch
+
+  @create_attrs %{
+    name: "some name",
+    contact_hours_per_week: Enum.random(20..48)
+  }
+  @update_attrs %{
+    name: "some updated name",
+    contact_hours_per_week: Enum.random(20..48)
+  }
+  @invalid_attrs %{
+    name: nil,
+    contact_hours_per_week: nil
+  }
+  @valid_fields [
+    "contact_hours_per_week",
+    "id",
+    "name"
+  ]
+
+  setup %{conn: conn} do
+    {:ok, conn: put_req_header(conn, "accept", "application/json")}
+  end
+
+  describe "index" do
+    test "lists all batches", %{conn: conn} do
+      conn = get(conn, Routes.batch_path(conn, :index))
+      [head | _tail] = json_response(conn, 200)
+      assert Map.keys(head) == @valid_fields
+    end
+  end
+
+  describe "create batch" do
+    test "renders batch when data is valid", %{conn: conn} do
+      conn = post(conn, Routes.batch_path(conn, :create), @create_attrs)
+      %{"id" => id} = json_response(conn, 201)
+
+      conn = get(conn, Routes.batch_path(conn, :show, id))
+
+      assert %{
+               "id" => ^id,
+               "name" => "some name",
+               "contact_hours_per_week" => contact_hours_per_week
+             } = json_response(conn, 200)
+
+      assert Enum.member?(20..48, contact_hours_per_week)
+    end
+
+    test "renders errors when data is invalid", %{conn: conn} do
+      conn = post(conn, Routes.batch_path(conn, :create), batch: @invalid_attrs)
+      assert json_response(conn, 422)["errors"] != %{}
+    end
+  end
+
+  describe "update batch" do
+    setup [:create_batch]
+
+    test "renders batch when data is valid", %{conn: conn, batch: %Batch{id: id} = batch} do
+      conn = put(conn, Routes.batch_path(conn, :update, batch), @update_attrs)
+      %{"id" => ^id} = json_response(conn, 200)
+
+      conn = get(conn, Routes.batch_path(conn, :show, id))
+
+      assert %{
+               "id" => ^id,
+               "name" => "some updated name",
+               "contact_hours_per_week" => contact_hours_per_week
+             } = json_response(conn, 200)
+
+      assert Enum.member?(20..48, contact_hours_per_week)
+    end
+
+    test "renders errors when data is invalid", %{conn: conn, batch: batch} do
+      conn = put(conn, Routes.batch_path(conn, :update, batch), @invalid_attrs)
+      assert json_response(conn, 422)["errors"] != %{}
+    end
+  end
+
+  describe "delete batch" do
+    setup [:create_batch]
```

Comment:
same here.
Also, why are we having `create_batch` under the `delete batch`

Reply:
In the provided code, the create_batch function is being used to create a batch record in the database before the delete batch test case is run. This is necessary because the delete batch test case expects a batch variable to be present in the context, which is used to specify the batch to be deleted.